src/commands/cmd_list.c

```
@@ -7,37 +7,89 @@
+		StubScanCtx sctx = {
+			.ctx       = ctx,
+			.stub_type = GraphStubType_Get (),
+			.n         = 0
+		};
 
-	// reply with each graph name
-	uint64_t     n   = 0;
-	GraphContext *gc = NULL;
+		RedisModuleScanCursor *cursor = RedisModule_ScanCursorCreate () ;
+		while (RedisModule_Scan (ctx, cursor, _collect_stub_name, &sctx)) ;
+		RedisModule_ScanCursorDestroy (cursor) ;
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)